🏗 Update how to execute jscodeshift

alanorozco opened the pull request:
I'm not sure if something has changed, but transforms I've run locally have failed lately due to configuration issues. Copied [default options](https://github.com/facebook/jscodeshift/blob/48f5d6d6e5e769639b958f1a955c83c68157a5fa/parser/babylon.js), plus `importAssertions`.

<!--
# Instructions:

- Pick a meaningful title for your pull request. (Use sentence case.)
  - Prefix the title with an emoji to identify what is being done. (Copy-paste the emoji from the list below.)
  - Do not overuse punctuation in the title (like `(chore):`).
  - If it is helpful, use a simple prefix (like `ProjectX: Implement some feature`).
- Enter a succinct description that says why the PR is necessary, and what it does.
  - Mention the GitHub issue that is being addressed by the pull request.
  - The keywords `Fixes`, `Closes`, or `Resolves` followed the issue number will automatically close the issue.

> NOTE: All non-trivial changes (like introducing new features or components) should have an associated issue or reference an I2I (intent-to-implement: go.amp.dev/i2i). Please read through the contribution process (go.amp.dev/contributing/code) for more information.

# Example of a good description:

- Implement aspect X
- Leave out feature Y because of A
- Improve performance by B
- Improve accessibility by C

# Emojis for categorizing pull requests (copy-paste emoji into description):

✨ New feature
🐛 Bug fix
🔥 P0 fix
✅ Tests
❄️ Flaky tests
🚀 Performance improvements
🖍 CSS / Styling
♿ Accessibility
🌐 Internationalization
📖 Documentation
🏗 Infrastructure / Tooling / Builds / CI
⏪ Reverting a previous change
♻️ Refactoring
🚮 Deleting code
🧪 Experimental code
-->

rsimha reviewed (approved):
LGTM with a question. (Feel free to punt if this is blocking development.)

Instead of explicitly copy-pasting the default configs here and having to maintain this file when things change, is there a way to say this config file extends whatever is default?

alanorozco commented:
> Instead of explicitly copy-pasting the default configs here and having to maintain this file when things change, is there a way to say this config file extends whatever is default?

No, I looked but couldn't find that. It doesn't seem like `jscodeshift`'s config resolution is as sophisticated as Babel's.

rsimha commented:
> No, I looked but couldn't find that. It doesn't seem like `jscodeshift`'s config resolution is as sophisticated as Babel's.

I see. This might've changed in a recent release: https://github.com/facebook/jscodeshift/releases

The reason we didn't detect it as a breaking change is because we haven't version locked `jscodeshift`, and are instead, invoking it via `npx`: 

https://github.com/ampproject/amphtml/blob/cde47367b9d802a3de0aa926704363e9a8521070/build-system/tasks/sweep-experiments/index.js#L101-L110

I'm not opposed to your adding it as a dependency and switching to the API instead of the CLI: https://www.npmjs.com/package/jscodeshift#the-jscodeshift-api. With this, we'll get a renovate PR for every new version, and we can preemptively detect and fix breaking changes.

alanorozco commented:
The [jscodeshift api](https://www.npmjs.com/package/jscodeshift#the-jscodeshift-api) refers to the object passed in a transform's implementation, not the Runner itself. The Runner doesn't seem to be intentionally accessible, [but some do use it](https://stackoverflow.com/questions/64770181/how-to-run-jscodeshift-from-node-api). However, it didn't work for me.

This shouldn't be an issue for versioning though. `npx jscodeshift` will use the local copy if added to `package.json`.

1. Added dependency to `package.json`.
2. Moved to `build-system/jscodeshift/parser-config.json` so it can be shared. (mainly with #32847)

rsimha commented on build-system/jscodeshift/README.md:
```diff
@@ -0,0 +1,7 @@
+# [jscodeshift](https://github.com/facebook/jscodeshift)
+
+This directory holds shared configuration and utils for **jscodeshift**.
```
I don't think this should be a top-level directory in `build-system/`. Since it has to do with configuration, can we move the entire `jscodeshift/` directory to `build-system/test-configs/jscodeshift/`?

rsimha reviewed (approved):
LGTM after moving the config dir into `test-configs/`. (See comment below.)

> This shouldn't be an issue for versioning though. `npx jscodeshift` will use the local copy if added to `package.json`

Excellent, this works too.
